taichi/transforms/offload.cpp

```diff
@@ -112,7 +112,8 @@ class Offloader {
           offloaded->const_end = true;
           offloaded->end_value = val->val.val_int32();
         } else {
-          if ((arch == Arch::opengl || arch == Arch::vulkan) &&
+          if ((arch == Arch::opengl || arch == Arch::vulkan ||
+               arch == Arch::gles) &&
```

Comment:
Just noiticing but shouldn't `dx11` also be here? Maybe this behavior should be covered under CompileConfig or a new struct called `CompilationStrategy`, and we will ask the backend (i.e. Program) to set a default set of parameters instead of putting `if` cases everywhere

Reply:
^ this should be put into a seperate issue